software/vcftools/main.nf

```diff
@@ -0,0 +1,129 @@
+    tuple val(meta), path("*.imiss"), optional:true, emit: missing_individual
+    tuple val(meta), path("*.lmiss"), optional:true, emit: missing_site
+    tuple val(meta), path("*.snpden"), optional:true, emit: snp_density
+    tuple val(meta), path("*.kept.sites"), optional:true, emit: kept_sites
+    tuple val(meta), path("*.removed.sites"), optional:true, emit: removed_sites
+    tuple val(meta), path("*.singletons"), optional:true, emit: singeltons
+    tuple val(meta), path("*.indel.hist"), optional:true, emit: indel_hist
+    tuple val(meta), path("*.hapcount"), optional:true, emit: hapcount
+    tuple val(meta), path("*.mendel"), optional:true, emit: mendel
+    tuple val(meta), path("*.FORMAT"), optional:true, emit: format
+    tuple val(meta), path("*.INFO"), optional:true, emit: info
+    tuple val(meta), path("*.012"), optional:true, emit: genotypes_matrix
+    tuple val(meta), path("*.012.indv"), optional:true, emit: genotypes_matrix_individual
+    tuple val(meta), path("*.012.pos"), optional:true, emit: genotypes_matrix_position
+    tuple val(meta), path("*.impute.hap"), optional:true, emit: impute_hap
+    tuple val(meta), path("*.impute.hap.legend"), optional:true, emit: impute_hap_legend
+    tuple val(meta), path("*.impute.hap.indv"), optional:true, emit: impute_hap_indv
+    tuple val(meta), path("*.ldhat.sites"), optional:true, emit: ldhat_sites
+    tuple val(meta), path("*.ldhat.locs"), optional:true, emit: ldhat_locs
+    tuple val(meta), path("*.BEAGLE.GL"), optional:true, emit: beagle_gl
+    tuple val(meta), path("*.BEAGLE.PL"), optional:true, emit: beagle_pl
+    tuple val(meta), path("*.ped"), optional:true, emit: ped
+    tuple val(meta), path("*.map"), optional:true, emit: map_
+    tuple val(meta), path("*.tped"), optional:true, emit: tped
+    tuple val(meta), path("*.tfam"), optional:true, emit: tfam
+    tuple val(meta), path("*.diff.sites_in_files"), optional:true, emit: diff_sites_in_files
+    tuple val(meta), path("*.diff.indv_in_files"), optional:true, emit: diff_indv_in_files
+    tuple val(meta), path("*.diff.sites"), optional:true, emit: diff_sites
+    tuple val(meta), path("*.diff.indv"), optional:true, emit: diff_indv
+    tuple val(meta), path("*.diff.discordance.matrix"), optional:true, emit: diff_discd_matrix
+    tuple val(meta), path("*.diff.switch"), optional:true, emit: diff_switch_error
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def args     = options.args.tokenize()
+
+    def bed_arg  = (options.args.contains('--bed')) ? "--bed ${bed}" :
+                   (options.args.contains('--exclude-bed')) ? "--exclude-bed ${bed}" :
+                   (options.args.contains('--hapcount')) ? "--hapcount ${bed}" : ''
+    args.removeIf { it.contains('--bed') }
+    args.removeIf { it.contains('--exclude-bed') }
+    args.removeIf { it.contains('--hapcount') }
```

Comment:
There's no way of using multiple of these options at the same time?

Reply:
That's a good point. Using --bed and --exclude-bed at the same time wouldn't really make sense (include and exclude sites) but i guess using the hapcount together with the earlier options could be a real scenario, although i'm not sure how common. One solution would be to add an additional optional input for this and separate it from this block. Alternatively this could be done in separate steps in the workflow context as is (maybe this is better for the moment?). I am unsure what's best. We already can't cover all use cases as they are just too broad - so we are relying on users being able to make small tweaks to best match their needs. But happy to look in to adding the aforementioned extra input channel to handle scenarios where someone wants to work with more than one bed file with a single command if you think that would be better?